### DISCUSSION TITLE
WSL Plugin: displaying error messages to the user

### DISCUSSION THREAD

**Discussion by ssijbabu** [microsoft/WSL#12369] [Q&A]
I am currently developing a WSL plugin that needs to validate the operating system of the distribution during the onDistributionStarted event. If the OS is not authorized, the plugin should:

- Display a message to the user: "You are not authorized to use this distribution".
- Return an error during the `onDistributionStarted` event to prevent further execution.

I’m reaching out to seek your guidance on how best to achieve this functionality. Specifically, I’d like to know:
- How to display the message to the user in the console and return an error from this event?

Any advice, examples, or resources you can share would be greatly appreciated. Thank you for your time and expertise!

**Comment by OneBlue** [comment]
@ssijbabu: You can use `PluginError` for that. 

For instance:

```

if (!DistributionNotAuthorized)
{
        g_api->PluginError(L"You are not authorized to use this distribution");
        return E_ACCESSDENIED;
}

```

Here's an [end to end example](https://github.com/microsoft/WSL/blob/5fd4ffb064ec3ab73c045e2bf86194c18180079f/test/windows/testplugin/Plugin.cpp#L59C1-L60C1)

Should do what you're looking for here.
